src/hooks/transaction/useCreateDepositMutation.ts

```diff
@@ -0,0 +1,31 @@
+import { useMutation } from 'react-query'
+import api from 'utils/api'
+
+type AddMoneyFormInput = {
+  id: string
+  note: string
+  amount: string
+  currency: string
+}
+
+const addMoney = async (formInput: AddMoneyFormInput) => {
+  return await api
+    .post(`accounts/${formInput.id}/transactions/add`, {
+      json: {
+        id: formInput.id,
+        note: formInput.note,
+        amount_money: {
+          amount: parseInt(formInput.amount),
+          currency: formInput.currency,
+        },
+      },
+    })
+    //201 HTTP code results in 'unexpected end of JSON input' error, so it must be text
+    .text()
+}
+
+const useCreateDepositMutation = () => {
+  return useMutation(addMoney)
```

Comment:
I will add an example of invalidating queries to this mutation:
Usually we invalidate queries when a mutation is successful. This will mean that when these queries are called again they will not show the cached value but make another API call.

The first invalidation is for the query to get the list of all transactions and the second is for the query to get a transaction by accountId. Notice how we will only invalidate the cache associated with this particular accountId. Searches with any other accountId will still be cached and displayed immediately.

```suggestion
  const queryClient = useQueryClient()

  return useMutation(addMoney, {
    onSuccess: (_, formInput) => {
      queryClient.invalidateQueries('transactions')
      queryClient.invalidateQueries(['transaction', formInput.id])
    },
  })
```